### PR TITLE
Gravity fix

### PR DESCRIPTION
**Pull request by wlsanderson**
fixed some of the math in how the direction and magnitude of gravity is applied to acceleration, flipped the sign convention of acceleration to be positive during motor burn as a default convention, and also checked flight data against the winter_2023_Gyro_Comp dataset.

### CODE REVIEW

File: airbrakes/data_handling/data_processor.py [hunk 1]
```diff
@@ -290,19 +310,15 @@ def _calculate_vertical_velocity(self) -> npt.NDArray[np.float64]:
         linear acceleration to get the speed.
         :return: A numpy array of the speed of the rocket at each data point
         """
-        # Get the vertical accelerations from the rotated acceleration vectors
+        # Gets the vertical accelerations from the rotated acceleration vectors. gravity needs to be
+        # subracted from vertical acceleration, Then deadbanded.
```
Comment (JacksonElia) [inline airbrakes/data_handling/data_processor.py:314]:
```suggestion
        # subtracted from vertical acceleration, Then deadbanded.
```

File: airbrakes/data_handling/data_processor.py [hunk 2]
```diff
@@ -290,19 +310,15 @@ def _calculate_vertical_velocity(self) -> npt.NDArray[np.float64]:
...
         vertical_accelerations = np.array(
             [
-                deadband(vertical_acceleration, ACCELERATION_NOISE_THRESHOLD)
+                deadband((vertical_acceleration - GRAVITY), ACCELERATION_NOISE_THRESHOLD)
```
Comment (JacksonElia) [inline airbrakes/data_handling/data_processor.py:317]:
```suggestion
                deadband(vertical_acceleration - GRAVITY, ACCELERATION_NOISE_THRESHOLD)
```

File: airbrakes/data_handling/data_processor.py [hunk 3]
```diff
@@ -132,13 +144,18 @@ def update(self, data_points: list[EstimatedDataPacket]) -> None:
                 ]
             )
             # TODO: Comment this line out before flight!
```
Comment (JacksonElia) [inline airbrakes/data_handling/data_processor.py:146]: Do you think you could add the gravity vectors to the csv files and remove this?

**Review by JacksonElia — COMMENTED** [review]
Lets merge this after you get the chance to update the csv files

**Comment by harshil21** [comment]
Ruff must pass before merging also.

**Comment by harshil21** [comment]
I'll be moving https://github.com/NCSU-High-Powered-Rocketry-Club/AirbrakesV2/pull/56/commits/32a8084812f8952aad7181ce16eb76db5512f1cf in a separate PR so we can check if the integration test passes on main

**Review by JacksonElia — APPROVED** [review]

**Pull request merged**